iceoryx_examples/singleprocess/single_process.cpp

```diff
@@ -44,39 +45,57 @@ void consoleOutput(const std::string& output)
 
 void sender()
 {
-    iox::popo::Publisher publisher({"Single", "Process", "Demo"});
+    iox::popo::UntypedPublisher publisher({"Single", "Process", "Demo"});
     publisher.offer();
 
     uint64_t counter{0};
     while (keepRunning.load())
     {
-        auto sample = static_cast<TransmissionData_t*>(publisher.allocateChunk(sizeof(TransmissionData_t)));
-        sample->counter = counter++;
-        consoleOutput(std::string("Sending: " + std::to_string(sample->counter)));
-        publisher.sendChunk(sample);
+        publisher.loan(sizeof(TransmissionData_t)).and_then([&](iox::popo::Sample<void>& sample) {
+            auto rawSample = static_cast<TransmissionData_t*>(sample.get());
+            rawSample->counter = counter++;
+            consoleOutput(std::string("Sending: " + std::to_string(rawSample->counter)));
+            sample.publish();
+        });
 
         std::this_thread::sleep_for(std::chrono::milliseconds(100));
     }
 }
 
 void receiver()
 {
-    iox::popo::Subscriber subscriber({"Single", "Process", "Demo"});
+    iox::popo::UntypedSubscriber subscriber({"Single", "Process", "Demo"});
```

Comment:
Oh, maybe all of our examples should use the typed publisher, except the ones showing how to use the untyped publisher of course.

With the typed publisher you don't need to get the raw pointer and cast it tot the actual type. With this, we teach our users how to use iceoryx in a safe way

Reply:
Using typed subscriber and direct access to sample!